orchagent/portsorch.cpp

```diff
@@ -3219,17 +3224,47 @@ bool PortsOrch::setBridgePortLearnMode(Port &port, string learn_mode)
     return true;
 }
 
-bool PortsOrch::addVlan(string vlan_alias)
+bool PortsOrch::addVlan(string vlan_alias, string vlan_type)
 {
     SWSS_LOG_ENTER();
 
     sai_object_id_t vlan_oid;
+    vector<sai_attribute_t> attrs;
 
     sai_vlan_id_t vlan_id = (uint16_t)stoi(vlan_alias.substr(4));
     sai_attribute_t attr;
     attr.id = SAI_VLAN_ATTR_VLAN_ID;
     attr.value.u16 = vlan_id;
-    sai_status_t status = sai_vlan_api->create_vlan(&vlan_oid, gSwitchId, 1, &attr);
+    attrs.push_back(attr);
+
+    /* SAI support is yet to be added for vlan_type attribute.
+     * For now, we can use SAI_VLAN_ATTR_META_DATA attr_id with
+     * hard-coded values. New enums yet to be introduced in SAI
+     * for values
+     *  0 = data (default)
+     *  1 = management
+     *  2 = control
+     *  Change this code once attribute and enums introduced in saivlan.h
+     *  */
+    if(vlan_type != "")
+    {
+        attr.id = SAI_VLAN_ATTR_META_DATA;
```

Comment:
What are the attributes here? Do you have a pointer to SAI proposal?. If it is not yet approved in SAI community, it will be too early for this changes.

Reply:
Done.